common/p_interaction.cpp

```diff
@@ -577,6 +577,85 @@ ItemEquipVal P_GivePower(player_t *player, int /*powertype_t*/ power)
 
 #include "v_textcolors.h"
 
+	/*
+ * @brief Player grabbed a resurrect player powerup
+ */
+static void P_ResurrectPlayerPowerUp(player_t* player)
+{
+	// Not lives game? Nothing to do.
+	if (!G_IsLivesGame())
+		return;
+
+	if (!player)
+		return;
+
+	// Grab all the players in game, make a list of their ids, then pick one at random
+	PlayersView ingame = PlayerQuery().notHasLives().execute().players;
+	std::vector<int> ingameplayers;
+	for (const auto& p : ingame)
```

Comment:
If only constructing this vector to check it's size, why not just do `int ingameplayers = std::count_if(ingame.begin(), ingame.end(), [&](const player_t* p){ return p->id != player.id })`

Reply:
It does more than check its size, its the list of eligible ids where the id is randomly selected.

Reply:
Am I reading something wrong? It looks like only the size is being used.

Reply:
I forgot to add where it actually selects from the array, whoops. Its coming soon